framework/src/base/NonlinearSystem.C

```diff
@@ -680,16 +694,28 @@ NonlinearSystem::computeResidual(NumericVector<Number> & residual, Moose::Kernel
 
   try
   {
-    residual.zero();
```

Comment:
Whoah - what's the reasoning for all of the change in logic here?

Now the residual vector _won't_ be zeroed if your just doing `KT_NONTIME`?  Nor will the `_time_integrator` get a `preStep()`?

Again, this looks like stuff specific to your usage... but not a good idea for MOOSE in general.